### PR TITLE
Rotate and flip

### PR DESCRIPTION
**Pull request by LoganTutt**
#### What issue does this fix?
- Issue #28 , Allows rotation and flipping of pieces in bullpen
#### Did you add tests to cover the code you propose?

Added unit test for rotation and flip moves

### CODE REVIEW

**Comment by PeterMitrano** [comment]
Mouse event and logic should be in controller, right?

**Comment by LoganTutt** [comment]
I could move the mouse event stuff to a controller, but its literally going to be copy pasting the stuff from the lambda to a new class. Since I dont think there are any other actions that the piece will have to do, Im not sure if a separate controller class is needed. Also, handling the right click menu is much simpler without having a separate controller

**Comment by PeterMitrano** [comment]
ok works for me

File: src/main/java/aeneas/models/Piece.java [hunk 1]
```diff
@@ -12,28 +12,65 @@
...
+      int row = s.getRow();
+      int col = s.getCol();
+      switch(direction) {
+      case CLOCKWISE:
+        s.setCol(-row+getHeight()-1);
+        s.setRow(col);
+        break;
+      case COUNTERCLOCKWISE:
+        s.setCol(row);
+        s.setRow(-col+getWidth()-1);
```
Comment (PeterMitrano) [inline src/main/java/aeneas/models/Piece.java:64]: you need a break at the end of this case so if someone wants to add another one they don't get a nasty surprise

File: src/main/java/aeneas/views/PieceView.java [hunk 2]
```diff
@@ -40,12 +66,117 @@ public PieceView(Piece pieceModel, int squareSize) {
...
+      PieceView fullSizedPieceView = new PieceView(pieceModel, model, BoardView.SQUARE_SIZE);
 
       Image snapshotImage = fullSizedPieceView.snapshot(snapshotParameters, null);
       db.setDragView(snapshotImage);
 
       event.consume();
     });
+    
+    
+    //callback for handling clicks on pieces
```
Comment (PeterMitrano) [inline src/main/java/aeneas/views/PieceView.java]: add documentation for what clicks do what

**Comment by PeterMitrano** [comment]
Use a VBox instead of a JFXListView
ListViews inherently scroll horizontally

**Pull request merged**